rb/lib/selenium/webdriver/common/proxy.rb

```diff
@@ -42,24 +42,23 @@ class Proxy
       def initialize(opts = {})
         opts = opts.dup
 
-        self.type           = opts.delete(:type) if opts.has_key? :type
-        self.ftp            = opts.delete(:ftp) if opts.has_key? :ftp
-        self.http           = opts.delete(:http) if opts.has_key? :http
-        self.no_proxy       = opts.delete(:no_proxy) if opts.has_key? :no_proxy
-        self.ssl            = opts.delete(:ssl) if opts.has_key? :ssl
-        self.pac            = opts.delete(:pac) if opts.has_key? :pac
-        self.auto_detect    = opts.delete(:auto_detect) if opts.has_key? :auto_detect
-        self.socks          = opts.delete(:socks) if opts.has_key? :socks
-        self.socks_username = opts.delete(:socks_username) if opts.has_key? :socks_username
-        self.socks_password = opts.delete(:socks_password) if opts.has_key? :socks_password
-
-        unless opts.empty?
-          raise ArgumentError, "unknown option#{'s' if opts.size != 1}: #{opts.inspect}"
-        end
+        self.type           = opts.delete(:type) if opts.key? :type
```

Comment:
Do we really need conditions in here? If `opts` does not have `:type`, calling `#type` will return `nil` (since instance variables are nils by default). `opts.delete(:type)` returns nil if there is no key, so we can assign it to `@type` - the logic does not change.